cartography/intel/azure/sql.py

```diff
@@ -277,267 +291,237 @@ def load_server_details(neo4j_session, credentials, subscription_id, details, up
+    MERGE (alias:AzureServerDNSAlias{id: dns_alias.id})
     ON CREATE SET alias.firstseen = timestamp(), alias.lastupdated = {azure_update_tag}
-    SET alias.name = {Name},
-    alias.dnsrecord = {AzureDNSRecord}
-    WITH alias
-    MATCH (s:AzureServer{id: {ServerId}})
+    SET alias.name = dns_alias.name,
+    alias.dnsrecord = dns_alias.azure_dns_record
+    WITH alias, dns_alias
+    MATCH (s:AzureSQLServer{id: dns_alias.server_id})
     MERGE (s)-[r:USED_BY]->(alias)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {azure_update_tag}
     """
 
-    for data in dns_aliases:
-        neo4j_session.run(
-            ingest_dns_aliases,
-            DNSAliasId=data['id'],
-            Name=data['name'],
-            AzureDNSRecord=get_optional_value(data, ['azure_dns_record']),
-            ServerId=data['server_id'],
-            azure_update_tag=update_tag,
-        )
+    neo4j_session.run(
+        ingest_dns_aliases,
+        dns_aliases_list=dns_aliases,
+        azure_update_tag=update_tag,
+    )
 
 
 @timeit
-def _load_server_ad_admins(neo4j_session, ad_admins, update_tag):
+def _load_server_ad_admins(
+        neo4j_session: neo4j.Session, ad_admins: List[Dict], update_tag: int
+) -> None:
     """
     Ingest the Server AD Administrators details into neo4j.
     """
     ingest_ad_admins = """
-    MERGE (a:AzureServerADAdministrator{id: {AdAdminId}})
+    UNWIND {ad_admins_list} as ad_admins
+    MERGE (a:AzureServerADAdministrator{id: ad_admins.id})
     ON CREATE SET a.firstseen = timestamp(), a.lastupdated = {azure_update_tag}
-    SET a.name = {Name},
-    a.type = {AdministratorType},
-    a.login = {Login}
-    WITH a
-    MATCH (s:AzureServer{id: {ServerId}})
+    SET a.name = ad_admins.name,
+    a.type = ad_admins.administrator_type,
+    a.login = ad_admins.login
+    WITH a, ad_admins
+    MATCH (s:AzureSQLServer{id: ad_admins.server_id})
     MERGE (s)-[r:ADMINISTERED_BY]->(a)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {azure_update_tag}
     """
 
-    for data in ad_admins:
-        neo4j_session.run(
-            ingest_ad_admins,
-            AdAdminId=data['id'],
-            Name=data['name'],
-            AdministratorType=data['administrator_type'],
-            Login=data['login'],
-            ServerId=data['server_id'],
-            azure_update_tag=update_tag,
-        )
+    neo4j_session.run(
+        ingest_ad_admins,
+        ad_admins_list=ad_admins,
+        azure_update_tag=update_tag,
+    )
 
 
 @timeit
-def _load_recoverable_databases(neo4j_session, recoverable_databases, update_tag):
+def _load_recoverable_databases(
+        neo4j_session: neo4j.Session, recoverable_databases: List[Dict], update_tag: int
+) -> None:
     """
     Ingest the recoverable database details into neo4j.
     """
     ingest_recoverable_databases = """
-    MERGE (rd:AzureRecoverableDatabase{id: {DatabaseId}})
+    UNWIND {recoverable_databases_list} as rec_dbs
```

Comment:
let's rename rec_dbs to rec_db, since it's a single object

Reply:
Done